firefox-wrapper: remove dead npapi plugin code

xaverdh opened the pull request:
###### Motivation for this change
Support for `npapi` plugins (except for flash) was dropped from `firefox` some time ago. None of the versions in `nixpkgs` actually supports them.
So remove support for them from the wrapper. 
 
###### Things done

* Verified that the hash does not change when building with default options
* Tested that building with one of the removed `nixpkgs` options throws an error as expected

xaverdh committed:
firefox-wrapper: remove dead npapi plugin code

xaverdh commented:
Most of the plugins in `pkgs/applications/networking/browsers/mozilla-plugins/` can probably be removed as well. Not sure if something else is still using them though.

worldofpeace commented on pkgs/applications/networking/browsers/firefox/wrapper.nix:
```diff
@@ -38,32 +35,25 @@ let
+            "enableAdobeReader"
+          ];
+        in if removed != []
+           then throw "Your configuration mentions ${lib.concatMapStringsSep ", " (p: browserName + "." + p) removed}. All plugin related options have been removed since Firefox no longer supports npapi plugins"
```
A link or a date would be nice.

xaverdh replied:
Will add a link

xaverdh commented:
After looking through `pkgs/applications/networking/browsers/mozilla-plugins/` and some use of grep, I am fairly sure that everything there except for the `flashplayer` is unused in nixpkgs. Should I add a commit to this pr for removing them, or over a new pr for that?

lheckemann commented:
IMHO add it to this PR.

xaverdh committed:
mozilla-plugins: remove all npapi plugins except for flash player

Firefox from version 52 onwards no longer supports npapi plugins (see https://support.mozilla.org/en-US/kb/npapi-plugins).

rycee commented:
Sorry, I missed this PR before. It would have been great if the `throw` could have been a `warn` for some period of time to avoid hard breaking of existing configurations.

xaverdh commented:
> Sorry, I missed this PR before. It would have been great if the `throw` could have been a `warn` for some period of time to avoid hard breaking of existing configurations.

You think its worth changing that still?

rycee commented:
I think it should be fine to keep the `throw` unless more complaints arise. I've removed the options from HM so people's configurations should now build OK again. But for future changes like this it would good to have a grace period during which a warning is printed instead of immediately going for a hard error.
